packages/binding-coap/src/coap-client.ts

```diff
@@ -140,29 +140,33 @@ export default class CoapClient implements ProtocolClient {
     });
   }
 
-  public subscribeResource(form: CoapForm, next: ((value: any) => void), error?: (error: any) => void, complete?: () => void): any {
-    let req = this.generateRequest(form, "GET", true);
+  public subscribeResource(form: CoapForm, next: ((value: any) => void), error?: (error: any) => void, complete?: () => void): Promise<Subscription> {
+    return new Promise<Subscription>((resolve, reject) => {
+      let req = this.generateRequest(form, "GET", true);
```

Comment:
Here we are doing a get not an observe... how is it working? on coaps we are correctly issuing an observe request.